multiversx_sdk/network_providers/proxy_network_provider.py

```diff
@@ -362,7 +364,19 @@ def do_post_generic(self, url: str, data: Any, url_parameters: Optional[dict[str
 
     def _do_get(self, url: str) -> GenericResponse:
         try:
-            response = requests.get(url, **self.config.requests_options)
+            retry_strategy = Retry(
+                total=self.config.requests_retry_options.retries,
+                backoff_factor=self.config.requests_retry_options.backoff_factor,
+                status_forcelist=self.config.requests_retry_options.status_forecelist,
+            )
+
+            adapter = HTTPAdapter(max_retries=retry_strategy)
+
+            with requests.Session() as session:
+                session.mount("https://", adapter)
+                session.mount("http://", adapter)
```

Comment:
A little bit of duplication. In the future, maybe we can extract something to `shared.py` - but only if it really makes sense.